internal/ingest/ingest.go

```diff
@@ -829,6 +839,16 @@ func (ing *Ingester) ingestWorkerLogic(provider peer.ID) {
 	// Filter out ads that are already processed, and any earlier ads.
 	splitAtIndex := len(assignment.adInfos)
 	for i, ai := range assignment.adInfos {
+		if ctx.Err() != nil {
+			log.Infow("Ingest worker canceled while ingesting ads", "provider", provider)
```

Comment:
log error too? Ditto in second check loop below.

Reply:
Done